### ISSUE TITLE
Support OpenAPI Links

### ISSUE THREAD

**Issue by DannyRyman** [RSuter/NSwag#1405]
I am correct in saying the OpenAPI links are current not implemented for the TypeScript and CSharp client generators? If not, are there any plans to support this in the future?

**Comment by RSuter** [comment]
Not implemented yet, not planned yet. Help would be really appreciated... the generators should be completely based on OpenApi 3

**Comment by RSuter** [comment]
There are varous parts involved to support this:

- [x] Add to openapi models (nswag.core)
- [ ] Automatically generate link info, based eg on reflection. How? (Custom attributes & operation processor)
- [ ] Generate typescript and csharp clients which expose these links

**Comment by rynowak** [comment]
No, we don't have any planned work in this area - no one has asked us for this, and it's not clear what kind of runtime feature's we'd add to support this.

**Comment by danilsonStan** [comment]
If I understand this correctly, the term link does not refer to a schema reference such as "$ref": "#/components/schemas/GeneralError". That would be really nice to have

**Comment by RSuter** [comment]
No link is metadata which describes linked data... $ref is already supported

**Comment by RSuter** [comment]
Ref: https://swagger.io/docs/specification/links/

@rynowak I think no runtime features would be needed - only metadata (e.g. attributes) to describe links to additional data for a given response, etc. that's why I think it doesn't have to be part of ASP.NET Core...

**Comment by skovalyova** [comment]
Hi @RicoSuter, are there any updates? Is it possible to use links as described in OpenAPI 3.0?
https://swagger.io/docs/specification/links/

**Comment by azan-n** [comment]
Hello @RicoSuter. Links are really interesting, especially with how they can create specialized clients in TS or test benches. If you could walk me through the code base briefly, I'll gladly submit a PR.
